sos/report/plugins/__init__.py

```diff
@@ -999,6 +1001,18 @@ def is_installed(self, package_name):
             len(self.policy.package_manager.all_pkgs_by_name(package_name)) > 0
         )
 
+    def check_snap_installed(self, package_name):
+        """Is the package $package_name installed and is a snap?
+
+        :param package_name:    The name of the package to check
+        :type package_name:     ``str``
+
+        :returns: ``True`` id the package is installed, else ``False``
+        :rtype: ``bool``
+        """
+        pkg = self.policy.package_manager.pkg_by_name(package_name)
+        self.is_snap = (pkg and pkg['pkg_manager'] == 'snap')
```

Comment:
yeah, I was pondering that idea before, and was trying different ideas. Agreed that we should mimic is_installed.

there could be a scenario where `self.policy.package_manager.pkg_by_name(package_name)` doesn't actually return anything, so testing for `pkg` is important, otherwise we'll get `keyErrror` with `pkg_manager` index not existing